.github/workflows/smb_acceptance.yml

```diff
@@ -0,0 +1,181 @@
+name: Acceptance
+
+# Optional, enabling concurrency limits: https://docs.github.com/en/actions/using-jobs/using-concurrency
+#concurrency:
+#  group: ${{ github.ref }}-${{ github.workflow }}
+#  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+# https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions
+permissions:
+  actions: none
+  checks: none
+  contents: none
+  deployments: none
+  id-token: none
+  issues: none
+  discussions: none
+  packages: none
+  pages: none
+  pull-requests: none
+  repository-projects: none
+  security-events: none
+  statuses: none
+
+on:
+  push:
+    branches-ignore:
+      - gh-pages
+      - metakitty
+  pull_request:
+    branches:
+      - '*'
+    paths:
+      - 'metsploit-framework.gemspec'
+      - 'Gemfile.lock'
+      - '**/**smb**'
+      - 'spec/acceptance/**'
+      - 'spec/support/acceptance/**'
+      - 'spec/acceptance_spec_helper.rb'
+#   Example of running as a cron, to weed out flaky tests
+#  schedule:
+#    - cron: '*/15 * * * *'
+
+jobs:
+  smb:
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 40
+
+    strategy:
+      fail-fast: true
+      matrix:
+        ruby:
+          - '3.2'
+        os:
+          - ubuntu-latest
+
+    env:
+      RAILS_ENV: test
+      SMB_USERNAME: acceptance_tests_user
+      SMB_PASSWORD: acceptance_tests_password
+
+    name: SMB Acceptance - ${{ matrix.os }} - Ruby ${{ matrix.ruby }}
+    steps:
+      - name: Install system dependencies
+        run: sudo apt-get install -y --no-install-recommends libpcap-dev graphviz smbclient samba
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Run Samba
+        working-directory: 'test/smb'
+        run: |
+          sudo useradd -m acceptance_tests_user
+          sudo cp -r shares /opt
+          sudo chmod -R 777 /opt/shares
+          sudo cp config/smb.conf /etc/samba/smb.conf
+          sudo smbpasswd -a acceptance_tests_user <<EOF
+          acceptance_tests_password
+          acceptance_tests_password
+          EOF
+          sudo smbpasswd -e acceptance_tests_user
+          sudo service smbd restart
```

Comment:
I'm confused; is this running directly on the host instead of being inside a docker container? 👀

Reply:
Fixed: Running on Docker now rather than runner itself. 👍